entc/gen/template/dialect/sql/query.tmpl

```diff
@@ -135,9 +135,7 @@ func ({{ $receiver }} *{{ $builder }}) sqlAll(ctx context.Context, hooks ...quer
 				{{- $fk1idx := 1 }}{{- $fk2idx := 0 }}{{ if $e.IsInverse }}{{ $fk1idx = 0 }}{{ $fk2idx = 1 }}{{ end }}
 				s.Join(joinT).On(s.C({{ $edgeid }}), joinT.C({{ $.Package }}.{{ $e.PKConstant }}[{{ $fk1idx }}]))
 				s.Where(sql.InValues(joinT.C({{ $.Package }}.{{ $e.PKConstant }}[{{ $fk2idx }}]), edgeIDs...))
-				columns := s.SelectedColumns()
-				s.Select(joinT.C({{ $.Package }}.{{ $e.PKConstant }}[{{ $fk2idx }}]))
-				s.AppendSelect(columns...)
+				s.PrependSelect(joinT.C({{ $.Package }}.{{ $e.PKConstant }}[{{ $fk2idx }}]))
 				s.SetDistinct(false)
```

Comment:
If there's a bug in the codegen, use the `entc/integration/integration_test.go` to write a test case that without this fix, the test fails. This will give us more context on the problem (if there's one) it's solved, and ensure we keep BC between versions.

Reply:
Added a integration test